src/main/java/internet/market/controllers/AddProductToShoppingCartController.java

```diff
@@ -23,7 +23,8 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
         String productId = req.getParameter("id");
         Long id = Long.valueOf(productId);
         shoppingCartService.addProduct(shoppingCartService
-                .getByUserId(USER_ID), productService.get(id));
+                .getByUserId((Long) req.getSession().getAttribute(USER_ID)),
```

Comment:
I would recommend to use local variables sometimes to make the code more readable